subsys/bluetooth/controller/hci_internal.h

```diff
@@ -24,6 +27,32 @@
  */
 int hci_internal_cmd_put(uint8_t *cmd_in);
 
+/** A user implementable HCI command handler
+ *
+ * @param[in]  cmd The HCI command itself
+ * @param[out] raw_event_out Parameters to be returned from the event
+ * @param[oun] param_length_out Length of parameters to be returned.
+ *
+ * @return Bluetooth status code. BT_HCI_ERR_UNKNOWN_CMD if unknown.
+ */
+typedef uint8_t (*hci_internal_user_cmd_handler_t)(uint8_t const * const cmd,
+						   uint8_t * const raw_event_out,
```

Comment:
does this need to be a command complete or command status event?

Reply:
It will always generate a command complete event. It is pointing to the return parameters of that event. 
I can add some more documentation related to this

Reply:
But what then if the command should generate a command status?

Reply:
That is not supported with these changes